lib/matplotlib/axes/_axes.py

```diff
@@ -4025,6 +4026,11 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
 
         if c_array is None:
             colors = c     # must be acceptable as PathCollection facecolors
+            try:
+                mcolors.to_rgba_array(colors)
+            except ValueError:
+                # c not acceptable as PathCollection facecolor
+                raise ValueError("c not acceptable as color sequence")
```

Comment:
This error message should probably contain the x shape, y shape, and c shape.  At this point we have tried all of the things the user could have meant, have not been able to work it out so we should give feedback on everything we tried.